src/base/bittorrent/session.cpp

```diff
@@ -321,6 +321,11 @@ QString Session::tempPath() const
     return m_tempPath;
 }
 
+QString Session::torrentTempPath(const InfoHash &hash) const
+{
+    return m_tempPath + QString(hash) + "/";
```

Comment:
>IIRC most modern filesystems has no limit for path names

You're right about the fs part, but not for Windows itself:
https://stackoverflow.com/questions/1880321/why-does-the-260-character-path-length-limit-exist-in-windows

Windows still has a limit about 260 bytes.

And I saw you used SHA-1 to hash it, you can encode the results to base64 then to string, it's shorter: 28 bytes, but then it might create collision on filesystems which are case insensitive...